src/tools/ceph_objectstore_tool.cc

```diff
@@ -1712,6 +1712,39 @@ int do_import_rados(string pool)
   return 0;
 }
 
+void filter_log(spg_t import_pgid, OSDMap &curmap, pg_log_t &log)
+{
+  list<pg_log_entry_t> oldlog;
+  oldlog.swap(log.log);
+
+  // XXX: Leaving head, tail, can_rollback_to, rollback_info_trimmed_to alone
+
+  for (list<pg_log_entry_t>::iterator i = oldlog.begin();
+       i != oldlog.end();
+       ) {
+
+    if (i->soid.nspace != g_ceph_context->_conf->osd_hit_set_namespace) {
+      object_t oid = i->soid.oid;
+      object_locator_t loc(i->soid);
+      // XXX: Do we need to set the hash? see also get_object()
+      // loc.hash = i->soid.get_hash();
+      pg_t raw_pgid = curmap.object_locator_to_pg(oid, loc);
+      pg_t pgid = curmap.raw_pg_to_pg(raw_pgid);
+
+      if (import_pgid.pgid == pgid) {
+        if (debug)
+          cerr << "Keep log entry for '" << i->soid << "'" << std::endl;
+        log.log.push_back(*i);
```

Comment:
As far as hitset object entries we decided to not import any hitsets, see get_object().  So if we don't import them, they shouldn't be in the log either.  I think we assumed hitsets become out of date anyway.